### PR TITLE
Static and Dynamic scoping

### LINKED ISSUE #6854: Static scoping & Dynamic Scoping
### Description

# Static scoping & Dynamic Scoping with example

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by vishnu95075**
<hr>

## Description 📜

Static scoping & Dynamic Scoping 
Fixes #6854

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

### CODE REVIEW

**Comment by vishnu95075** [comment]
@gunjan-g  please give example of Add index line in readme along with relative link of folder created

File: C_CPP/Fundamentals/Static scoping & Dynamic Scoping/Static scoping & Dynamic Scoping.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	printf("%d",fun2());
+	return 0;
+}
+```
+Output
+```
+10
+```
+In this case compiler first search in the current block then after global. So output is here **10**.
+
```
Comment (gunjan-g) [inline C_CPP/Fundamentals/Static scoping & Dynamic Scoping/Static scoping & Dynamic Scoping.md:53]:
```suggestion
<br>
```

File: C_CPP/Fundamentals/Static scoping & Dynamic Scoping/readme.md [hunk 2]
```diff
@@ -1 +1 @@
-
+- Static and Dynamic scoping
```
Comment (gunjan-g) [inline C_CPP/Fundamentals/Static scoping & Dynamic Scoping/readme.md:1]:
```suggestion
- Static and Dynamic scoping (./Static scoping & Dynamic Scoping.md)
```

**Review by gunjan-g — CHANGES_REQUESTED** [review]
@vishnu95075 Do the required changes

File: C_CPP/Fundamentals/Static scoping & Dynamic Scoping/readme.md (hunk 2 above)
Comment (gunjan-g) [reply C_CPP/Fundamentals/Static scoping & Dynamic Scoping/readme.md]: @vishnu95075 Just copy and paste the above suggested index line, you have to add this relative link

**Comment by vishnu95075** [comment]
@Saviour1001 thanks for suggestion. I am resolve that please review.

**Comment by Saviour1001** [comment]
Looks perfect !

**Pull request merged**